packages/block-library/src/navigation/index.php

```diff
@@ -209,10 +209,22 @@ function block_core_navigation_build_html( $attributes, $block, $colors, $font_s
 			'<a class="wp-block-navigation-link__content"';
 
 		// Start appending HTML attributes to anchor tag.
-		if ( isset( $block['attrs']['url'] ) ) {
+		if (
+			isset( $block['attrs']['url'] ) &&
+			isset( $block['attrs']['type'] ) &&
+			'URL' === $block['attrs']['type']
+		) {
 			$html .= ' href="' . esc_url( $block['attrs']['url'] ) . '"';
 		}
 
+		if (
+			isset( $block['attrs']['type'] ) &&
+			isset( $block['attrs']['id'] ) &&
+			'URL' !== $block['attrs']['type']
+		) {
+			$html .= ' href="' . get_permalink( $block['attrs']['id'] ) . '"';
```

Comment:
Checking through core code, `get_permalink` seems to be often escaped using `esc_url` when used in markup though sometimes not.

I think best to stay on the side of caution and escape it.